modules/siddhi-query-api/src/main/java/org/wso2/siddhi/query/api/expression/Expression.java

```diff
@@ -234,5 +239,67 @@ public static TimeConstant year(int i) {
             return year((long) i);
         }
 
+        public static Long timeToLong(String value) {
+            Pattern timeValuePattern = Pattern.compile("\\d+");
+            Pattern durationPattern = Pattern.compile("\\D+");
+            Matcher timeMatcher = timeValuePattern.matcher(value);
+            Matcher durationMatcher = durationPattern.matcher(value);
+            int timeValue;
+            TimePeriod.Duration duration;
+            if (timeMatcher.find() && durationMatcher.find()) {
+                duration = normalizeDuration(durationMatcher.group(0).trim());
+                timeValue = Integer.parseInt(timeMatcher.group(0));
+                try {
+                    switch (duration) {
+                        case SECONDS:
+                            return Expression.Time.sec(timeValue).value();
+                        case MINUTES:
+                            return Expression.Time.minute(timeValue).value();
+                        case HOURS:
+                            return Expression.Time.hour(timeValue).value();
+                        case DAYS:
+                            return Expression.Time.day(timeValue).value();
+                        case YEARS:
+                            return Expression.Time.year(timeValue).value();
+                        default:
+                            return Expression.Time.month(timeValue).value();
+                    }
+                } catch (NoSuchDurationException e) {
+                    throw new NoSuchDurationException("Duration" + timeValue + "does not exists", e);
+                }
```

Comment:
Why are we catching NoSuchDurationException here and throwing it again? What is throwing NoSuchDurationException here?